### ISSUE TITLE
BadParcelableException when destination accepts interface argument

### ISSUE THREAD

**Issue by pchmielowski** [raamcosta/compose-destinations#60]
Hi! I've encountered a following exception:

```
    android.os.BadParcelableException: java.lang.NoSuchFieldException: CREATOR
        at com.ramcosta.composedestinations.navargs.parcelable.DefaultParcelableNavTypeSerializer.getParcelableCreator(DefaultParcelableNavTypeSerializer.kt:58)
        at com.ramcosta.composedestinations.navargs.parcelable.DefaultParcelableNavTypeSerializer.base64ToParcelable(DefaultParcelableNavTypeSerializer.kt:39)
        at com.ramcosta.composedestinations.navargs.parcelable.DefaultParcelableNavTypeSerializer.fromRouteString(DefaultParcelableNavTypeSerializer.kt:24)
        at net.chmielowski.pill.ui.navtype.ArgumentNavType.parseValue(ArgumentNavType.kt:26)
        at net.chmielowski.pill.ui.navtype.ArgumentNavType.parseValue(ArgumentNavType.kt:13)
        at androidx.navigation.NavType.parseAndPut(NavType.kt:80)
        at androidx.navigation.NavDeepLink.parseArgument(NavDeepLink.kt:228)
        at androidx.navigation.NavDeepLink.getMatchingArguments(NavDeepLink.kt:173)
        at androidx.navigation.NavDestination.matchDeepLink(NavDestination.kt:347)
        at androidx.navigation.NavGraph.matchDeepLink(NavGraph.kt:72)
        at androidx.navigation.NavController.navigate(NavController.kt:1648)
        at androidx.navigation.NavController.navigate(NavController.kt:1980)
        at androidx.navigation.NavController.navigate$default(NavController.kt:1975)
        at androidx.navigation.NavController.navigate(NavController.kt:1961)
        at com.ramcosta.composedestinations.navigation.DestinationsNavController.navigate(DestinationsNavController.kt:27)
        at com.ramcosta.composedestinations.navigation.DestinationsNavigator$DefaultImpls.navigate(DestinationsNavigator.kt:40)
        at com.ramcosta.composedestinations.navigation.DestinationsNavController.navigate(DestinationsNavController.kt:13)
        at com.ramcosta.composedestinations.navigation.DestinationsNavigator$DefaultImpls.navigate$default(DestinationsNavigator.kt:35)
```

It happens right after clicking the button, when `navigate()` is called:
```kotlin
@Composable
@Destination(start = true)
fun First(navigator: DestinationsNavigator) {
    Button(onClick = { navigator.navigate(SecondDestination(ArgumentImpl(2137))) }) {
        Text("->")
    }
}

@Composable
@Destination
fun Second(argument: Argument) {
}

interface Argument : Parcelable

@Parcelize
data class ArgumentImpl(val value: Int) : Argument
```

When I change the signature of `Second` function to: `fun Second(argument: ArgumentImpl)`, everything works correctly and there is no exception.

I'm using following versions:
```
kotlin_version = "1.6.10"

id 'com.google.devtools.ksp' version "$kotlin_version-1.0.2"

final destinations_version = "1.3.1-beta"
implementation "io.github.raamcosta.compose-destinations:core:$destinations_version"
ksp "io.github.raamcosta.compose-destinations:ksp:$destinations_version"
```

Let me know if you need more information on this issue, I'd be happy to help!

**Comment by raamcosta** [comment]
Hi @pchmielowski !
Thanks for reporting.

Interesting I haven't thought of this before 🤔 
I know where is the problem. I will investigate this hopefully today still 🙂 

I'll keep you posted.

**Comment by pchmielowski** [comment]
Great news!

**Comment by raamcosta** [comment]
Fix released today with version 1.3.2. Sorry it took me longer than expected, it did not seem to urgent 😅 
Hope it was not blocking you!

**Comment by pchmielowski** [comment]
It was my personal project and I've temporarily used `Serializable` instead of `Parcelable` - but it's a great that now I can upgrade and refactor! Thanks!
